data/mapcss-mapping.csv

```diff
@@ -689,18 +689,18 @@ place|continent;601;
 deprecated|deprecated;602;x
 deprecated|deprecated;603;x
 boundary|administrative|nation;[boundary=administrative][border_type=nation];;name;int_name;604;
-deprecated|deprecated;605;x
-deprecated|deprecated;606;x
-deprecated|deprecated;607;x
+piste:type|hike;605;
+piste:type|connection;606;
+piste:type|skitour;607;
 aerialway|magic_carpet;608;
-deprecated|deprecated;609;x
-deprecated|deprecated;610;x
-deprecated|deprecated;611;x
-deprecated|deprecated;612;x
-deprecated|deprecated;613;x
-deprecated|deprecated;614;x
-deprecated|deprecated;615;x
-deprecated|deprecated;616;x
+piste:type|downhill|novice|area;[piste:type=downhill][piste:difficulty=novice][area?];;;;609;
+piste:type|downhill|easy|area;[piste:type=downhill][piste:difficulty=easy][area?];;;;610;
+piste:type|downhill|intermediate|area;[piste:type=downhill][piste:difficulty=intermediate][area?];;;;611;
+piste:type|downhill|advanced|area;[piste:type=downhill][piste:difficulty=advanced][area?];;;;612;
+piste:type|downhill|expert|area;[piste:type=downhill][piste:difficulty=expert][area?];;;;613;
+piste:type|downhill|area;[piste:type=downhill][area?];;;;614;
+piste:type|sled|area;[piste:type=sled][area?];;;;615;
```

Comment:
Do we need separate area types?
I suppose we can define line and area rules for the initial piste:type|downhill|novice type and everything should be ok, no?

Reply:
Then areas will have a border around them (doesn't look good IMO, especially when two areas connect to each other there will be a line separating them) and we won't have an option to render just centerlines without areas (like it is now in the main style).

Reply:
Hm, should we fix it? Like use line rules for line objects only, and area rules for area objects only?

We don't draw _the center line_ for area objects anyway.

Reply:
I don't like _generating_ new types. We crutch drawing but lose in some other processing (like routing or search).

Reply:
This should be discussed and implemented separately in conjunction with #5884 and other cases like dams and some barriers..

Reply:
> I don't like _generating_ new types. We crutch drawing but lose in some other processing (like routing or search).

This particular PR doesn't impact search or routing anyhow, right?

Reply:
https://github.com/organicmaps/organicmaps/issues/7075